src/otx/v2/adapters/torch/lightning/visual_prompt/dataset.py

```diff
@@ -0,0 +1,171 @@
+        data_format: str | None = None,
+    ) -> None:
+        """Initialize a Dataset object for VISUAL_PROMPTING tasks.
+
+        Args:
+            task (Optional[Union[TaskType, str]]): The type of VISUAL_PROMPTING task to perform.
+                Defaults to None.
+            train_type (Optional[Union[TrainType, str]]): The type of training to perform.
+                Defaults to None.
+            train_data_roots (Optional[str]): The path to the training data root directory.
+                Defaults to None.
+            train_ann_files (Optional[str]): The path to the training annotation file.
+                Defaults to None.
+            val_data_roots (Optional[str]): The path to the validation data root directory.
+                Defaults to None.
+            val_ann_files (Optional[str]): The path to the validation annotation file.
+                Defaults to None.
+            test_data_roots (Optional[str]): The path to the test data root directory.
+                Defaults to None.
+            test_ann_files (Optional[str]): The path to the test annotation file.
+                Defaults to None.
+            unlabeled_data_roots (Optional[str]): The path to the unlabeled data root directory.
+                Defaults to None.
+            unlabeled_file_list (Optional[str]): The path to the unlabeled file list.
+                Defaults to None.
+            data_format (Optional[str]): The format of the data. Defaults to "mvtec".
+
+        Returns:
+            None
+        """
+        super().__init__(
+            task,
+            train_type,
+            train_data_roots,
+            train_ann_files,
+            val_data_roots,
+            val_ann_files,
+            test_data_roots,
+            test_ann_files,
+            unlabeled_data_roots,
+            unlabeled_file_list,
+            data_format=data_format,
+        )
+
+    def build_dataset(
+        self,
+        subset: str,
+        pipeline: dict | list | None = None,  # transform_config
+        config: str | (DictConfig | dict) | None = None,
+    ) -> TorchDataset | None:
+        """Build a TorchDataset for the given subset using the specified pipeline and configuration.
+
+        Args:
+            subset (str): The subset to build the dataset for.
+            pipeline (dict | list | None, optional): The pipeline to use for data transformation.
+            config (str | (DictConfig | dict) | None, optional): The configuration to use for the dataset.
+
+        Returns:
+            Optional[TorchDataset]: The built TorchDataset, or None if the dataset is empty.
+        """
+        if not self.initialize:
+            self._initialize()
+
+        config = OmegaConf.load(filename=config) if isinstance(config, str) else DictConfig({})
+        config = config.get("dataset", config)
+        image_size = config.get("image_size", 1024)
+        normalize = config.get("normalize", {})
```

Comment:
In the mmXdataset, we used `get_default_pipeline` function for this initialization. is there reason to select this way for the lightening case?

Reply:
We might use https://github.com/openvinotoolkit/training_extensions/pull/2583/files#diff-3d5379cce455de27aa201ecffd46130371bd4639d69c431a96aee05a887df4a2R59

Reply:
Maybe Lightning will be a different way, I'll revisit the pipeline later.